src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -466,26 +473,29 @@ def _user_confirmed(confirmation, command_args):
         return False
 
 
-def _polish_rp_not_registerd_error(cli_error):
-    msg = str(cli_error)
-    pertinent_text_namespace = 'The subscription must be registered to use namespace'
-    pertinent_text_feature = 'is not registered for feature'
-    # pylint: disable=line-too-long
-    if pertinent_text_namespace in msg:
-        reg = r".*{} '(.*)'".format(pertinent_text_namespace)
-        match = re.match(reg, msg)
-        cli_error = CLIError(
-            "Run 'az provider register -n {}' to register the namespace first".format(
-                match.group(1)))
-    elif pertinent_text_feature in msg:
-        reg = r".*{}\s+([^\s]+)\s+".format(pertinent_text_feature)
```

Comment:
`feature` is much less used and sniffing the text is a bit tricky i like to avoid, so i am removing the polish rather focus on the provider error, which end customers do hit.